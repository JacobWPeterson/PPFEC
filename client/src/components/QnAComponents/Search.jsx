import React from 'react';
import styled from 'styled-components';

const Input = styled.input.attrs(() => ({
  type: 'text',
}))`
  border-radius: 1px;
  border-color: black;
  width: 66vw;
  z-index: 0;
  position: relative;
  padding: 1.5vh 2vw;
  
`;

const Wrapper = styled.div`
  width: 100%;
`;

const Magnify = styled.div`
  text-align: right;
  z-index: 50;
  position: relative;
  top: -5.5vh;
  left: 66vw;
  width: 1rem;
  padding-right: 4vw;
  color: black;
  -webkit-transform: rotate(-45deg); 
  -moz-transform: rotate(-45deg); 
  -o-transform: rotate(-45deg);
  transform: rotate(-45deg);
`;

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({ input: e.target.value }, () => {
      if (this.state.input.length > 2) {
        this.props.func(this.state.input);
      } else {
        this.props.reset();
      }
    });
  }

  render() {
    return (
      <Wrapper>
        <Input value={this.state.input} onChange={this.handleChange} placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..." />
        <Magnify>âš²</Magnify>
      </Wrapper>
    );
  }
}
export default Search;
