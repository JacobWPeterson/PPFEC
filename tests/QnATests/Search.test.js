import React from 'react';
import { mount } from 'enzyme';
import Search from '../../client/src/components/QnAComponents/Search.jsx';

describe('One Search Test', () => {
  const SearchComp = mount(<Search debug />);
  const SearchBody = (SearchComp.find('div').at(0).text());

  it('Should render correctly', () => {
    expect(SearchComp).toMatchSnapshot();
  });

  it('Should render the Search body correctly', () => {
    expect(SearchBody).toEqual('âš²');
  });
});
