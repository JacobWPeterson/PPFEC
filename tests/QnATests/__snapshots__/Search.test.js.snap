// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One Search Test Should render correctly 1`] = `
<Search
  debug={true}
  reset={[Function]}
>
  <Search__Wrapper>
    <div
      className="Search__Wrapper-np7a72-1 biulzn"
    >
      <Search__Input
        onChange={[Function]}
        placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
        type="text"
        value=""
      >
        <input
          className="Search__Input-np7a72-0 iZcJlO"
          onChange={[Function]}
          placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
          type="text"
          value=""
        />
      </Search__Input>
      <Search__Magnify>
        <div
          className="Search__Magnify-np7a72-2 jCFwuN"
        >
          âš²
        </div>
      </Search__Magnify>
    </div>
  </Search__Wrapper>
</Search>
`;
